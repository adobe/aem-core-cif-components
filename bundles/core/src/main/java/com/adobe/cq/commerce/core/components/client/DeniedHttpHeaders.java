/*
 *  Copyright 2021 Adobe. All rights reserved.
 *
 *   This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */

package com.adobe.cq.commerce.core.components.client;

import java.util.Set;

import com.google.common.collect.ImmutableSet;

public interface DeniedHttpHeaders {
    /**
     * A list of HTTP headers that cannot be overriden when configuring a list of custom HTTP headers
     */
    final Set<String> DENYLIST = ImmutableSet.of(org.apache.http.HttpHeaders.ACCEPT,
        org.apache.http.HttpHeaders.ACCEPT_CHARSET,
        org.apache.http.HttpHeaders.ACCEPT_ENCODING,
        org.apache.http.HttpHeaders.ACCEPT_LANGUAGE,
        org.apache.http.HttpHeaders.ACCEPT_RANGES,
        org.apache.http.HttpHeaders.AGE,
        org.apache.http.HttpHeaders.ALLOW,
        org.apache.http.HttpHeaders.AUTHORIZATION,
        org.apache.http.HttpHeaders.CACHE_CONTROL,
        org.apache.http.HttpHeaders.CONNECTION,
        org.apache.http.HttpHeaders.CONTENT_ENCODING,
        org.apache.http.HttpHeaders.CONTENT_LANGUAGE,
        org.apache.http.HttpHeaders.CONTENT_LENGTH,
        org.apache.http.HttpHeaders.CONTENT_LOCATION,
        org.apache.http.HttpHeaders.CONTENT_MD5,
        org.apache.http.HttpHeaders.CONTENT_RANGE,
        org.apache.http.HttpHeaders.CONTENT_TYPE,
        org.apache.http.HttpHeaders.DATE,
        org.apache.http.HttpHeaders.DAV,
        org.apache.http.HttpHeaders.DEPTH,
        org.apache.http.HttpHeaders.DESTINATION,
        org.apache.http.HttpHeaders.ETAG,
        org.apache.http.HttpHeaders.EXPECT,
        org.apache.http.HttpHeaders.EXPIRES,
        org.apache.http.HttpHeaders.FROM,
        org.apache.http.HttpHeaders.HOST,
        org.apache.http.HttpHeaders.IF,
        org.apache.http.HttpHeaders.IF_MATCH,
        org.apache.http.HttpHeaders.IF_MODIFIED_SINCE,
        org.apache.http.HttpHeaders.IF_NONE_MATCH,
        org.apache.http.HttpHeaders.IF_RANGE,
        org.apache.http.HttpHeaders.IF_UNMODIFIED_SINCE,
        org.apache.http.HttpHeaders.LAST_MODIFIED,
        org.apache.http.HttpHeaders.LOCATION,
        org.apache.http.HttpHeaders.LOCK_TOKEN,
        org.apache.http.HttpHeaders.MAX_FORWARDS,
        org.apache.http.HttpHeaders.OVERWRITE,
        org.apache.http.HttpHeaders.PRAGMA,
        org.apache.http.HttpHeaders.PROXY_AUTHENTICATE,
        org.apache.http.HttpHeaders.PROXY_AUTHORIZATION,
        org.apache.http.HttpHeaders.RANGE,
        org.apache.http.HttpHeaders.REFERER,
        org.apache.http.HttpHeaders.RETRY_AFTER,
        org.apache.http.HttpHeaders.SERVER,
        org.apache.http.HttpHeaders.STATUS_URI,
        org.apache.http.HttpHeaders.TE,
        org.apache.http.HttpHeaders.TIMEOUT,
        org.apache.http.HttpHeaders.TRAILER,
        org.apache.http.HttpHeaders.TRANSFER_ENCODING,
        org.apache.http.HttpHeaders.UPGRADE,
        org.apache.http.HttpHeaders.USER_AGENT,
        org.apache.http.HttpHeaders.VARY,
        org.apache.http.HttpHeaders.VIA,
        org.apache.http.HttpHeaders.WARNING,
        org.apache.http.HttpHeaders.WWW_AUTHENTICATE,
        "Store",
        "Preview-Version");
}
