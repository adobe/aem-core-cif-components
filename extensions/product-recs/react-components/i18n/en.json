{
  "productrecs:add-to-cart": [
    {
      "type": 0,
      "value": "Add to Cart"
    }
  ],
  "productrecs:add-to-wishlist": [
    {
      "type": 0,
      "value": "Add to Wish List"
    }
  ],
  "productrecs:price-from": [
    {
      "type": 0,
      "value": "from"
    }
  ]
}