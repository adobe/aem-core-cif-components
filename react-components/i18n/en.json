{
  "account:account-created": [
    {
      "type": 0,
      "value": "Account created"
    }
  ],
  "account:account-created-title": [
    {
      "type": 0,
      "value": "Your account was successfully created"
    }
  ],
  "account:account-details-sign-in-text": [
    {
      "type": 0,
      "value": "Please sign in to see the account details."
    }
  ],
  "account:account-icon-text-greeting": [
    {
      "type": 0,
      "value": "Hi, "
    },
    {
      "type": 1,
      "value": "name"
    }
  ],
  "account:account-icon-text-sign-in": [
    {
      "type": 0,
      "value": "Sign In"
    }
  ],
  "account:account-information": [
    {
      "type": 0,
      "value": "Account Information"
    }
  ],
  "account:add-an-address": [
    {
      "type": 0,
      "value": "Add an address"
    }
  ],
  "account:address-book": [
    {
      "type": 0,
      "value": "Address Book"
    }
  ],
  "account:address-book-sign-in-text": [
    {
      "type": 0,
      "value": "Please Sign in to see your address book."
    }
  ],
  "account:address-default-tag": [
    {
      "type": 0,
      "value": "Default"
    }
  ],
  "account:address-delete": [
    {
      "type": 0,
      "value": "Delete"
    }
  ],
  "account:address-delete-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:address-delete-confirm": [
    {
      "type": 0,
      "value": "Delete"
    }
  ],
  "account:address-edit": [
    {
      "type": 0,
      "value": "Edit"
    }
  ],
  "account:address-form-heading": [
    {
      "type": 0,
      "value": "Address"
    }
  ],
  "account:change-password": [
    {
      "type": 0,
      "value": "Change Password"
    }
  ],
  "account:change-password-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:confirm-new-password": [
    {
      "type": 0,
      "value": "Confirm New Password"
    }
  ],
  "account:confirm-password": [
    {
      "type": 0,
      "value": "Confirm Password"
    }
  ],
  "account:continue-shopping": [
    {
      "type": 0,
      "value": "Continue Shopping"
    }
  ],
  "account:create-account": [
    {
      "type": 0,
      "value": "Create account"
    }
  ],
  "account:create-account-lead": [
    {
      "type": 0,
      "value": "Check out faster, use multiple addresses, track orders and more by creating an account!"
    }
  ],
  "account:create-an-account": [
    {
      "type": 0,
      "value": "Create an Account"
    }
  ],
  "account:create-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:create-submit": [
    {
      "type": 0,
      "value": "Submit"
    }
  ],
  "account:current-password": [
    {
      "type": 0,
      "value": "Current Password"
    }
  ],
  "account:edit-account-info": [
    {
      "type": 0,
      "value": "Edit account information"
    }
  ],
  "account:email": [
    {
      "type": 0,
      "value": "Email address"
    }
  ],
  "account:email-confirmation-info": [
    {
      "type": 0,
      "value": "You will receive a link at "
    },
    {
      "type": 1,
      "value": "email"
    },
    {
      "type": 0,
      "value": ". Access that link to confirm your email address."
    }
  ],
  "account:firstname": [
    {
      "type": 0,
      "value": "First Name"
    }
  ],
  "account:forget-password-confirmation": [
    {
      "type": 0,
      "value": "If there is an account associated with "
    },
    {
      "type": 1,
      "value": "email"
    },
    {
      "type": 0,
      "value": ", you will receive an email with a link to change your password."
    }
  ],
  "account:forgot-password": [
    {
      "type": 0,
      "value": "Forgot Password?"
    }
  ],
  "account:forgot-password-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:forgot-password-submit": [
    {
      "type": 0,
      "value": "Submit"
    }
  ],
  "account:lastname": [
    {
      "type": 0,
      "value": "Last Name"
    }
  ],
  "account:my-account": [
    {
      "type": 0,
      "value": "My account"
    }
  ],
  "account:name": [
    {
      "type": 0,
      "value": "Name"
    }
  ],
  "account:new-password": [
    {
      "type": 0,
      "value": "New Password"
    }
  ],
  "account:new-password-confirm": [
    {
      "type": 0,
      "value": "Confirm new Password"
    }
  ],
  "account:password": [
    {
      "type": 0,
      "value": "Password"
    }
  ],
  "account:password-recovery": [
    {
      "type": 0,
      "value": "Password recovery"
    }
  ],
  "account:password-was-changed": [
    {
      "type": 0,
      "value": "Your password was changed."
    }
  ],
  "account:reset-password": [
    {
      "type": 0,
      "value": "Reset Password"
    }
  ],
  "account:reset-password-error": [
    {
      "type": 0,
      "value": "Could not reset password."
    }
  ],
  "account:reset-password-instructions": [
    {
      "type": 0,
      "value": "Enter your email below to receive a password reset link."
    }
  ],
  "account:reset-password-lead": [
    {
      "type": 0,
      "value": "Please choose a new password to complete the password reset."
    }
  ],
  "account:reset-password-missing-token": [
    {
      "type": 0,
      "value": "Missing or invalid token."
    }
  ],
  "account:reset-password-success": [
    {
      "type": 0,
      "value": "Your password was changed. Please log in with your new password."
    }
  ],
  "account:sign-in": [
    {
      "type": 0,
      "value": "Sign In"
    }
  ],
  "account:sign-in-form-title": [
    {
      "type": 0,
      "value": "Sign-in to Your Account"
    }
  ],
  "account:sign-out": [
    {
      "type": 0,
      "value": "Sign Out"
    }
  ],
  "account:signing-in": [
    {
      "type": 0,
      "value": "Signing In"
    }
  ],
  "account:subscribe-news": [
    {
      "type": 0,
      "value": "Subscribe to news and updates"
    }
  ],
  "cart:apply-coupon": [
    {
      "type": 0,
      "value": "Apply Coupon"
    }
  ],
  "cart:continue-shopping": [
    {
      "type": 0,
      "value": "Continue Shopping"
    }
  ],
  "cart:edit-item": [
    {
      "type": 0,
      "value": "Edit item"
    }
  ],
  "cart:enter-coupon": [
    {
      "type": 0,
      "value": "Enter your code"
    }
  ],
  "cart:fetching-data": [
    {
      "type": 0,
      "value": "Fetching cart data..."
    }
  ],
  "cart:itemsWithCount": [
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 7
            },
            {
              "type": 0,
              "value": " item"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 7
            },
            {
              "type": 0,
              "value": " items"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "cart:new-total": [
    {
      "type": 0,
      "value": "New Cart Total:"
    }
  ],
  "cart:no-items": [
    {
      "type": 0,
      "value": "There are no items in your shopping cart"
    }
  ],
  "cart:quantity": [
    {
      "type": 0,
      "value": "Quantity"
    }
  ],
  "cart:remove-coupon": [
    {
      "type": 0,
      "value": "Remove coupon"
    }
  ],
  "cart:remove-item": [
    {
      "type": 0,
      "value": "Remove item"
    }
  ],
  "cart:shopping-cart": [
    {
      "type": 0,
      "value": "Shopping Cart"
    }
  ],
  "cart:total": [
    {
      "type": 0,
      "value": "Cart Total:"
    }
  ],
  "cart:update-cart": [
    {
      "type": 0,
      "value": "Update Cart"
    }
  ],
  "checkout:add-billing-information": [
    {
      "type": 0,
      "value": "Add Billing Information"
    }
  ],
  "checkout:add-shipping-information": [
    {
      "type": 0,
      "value": "Add Shipping Information"
    }
  ],
  "checkout:address-city": [
    {
      "type": 0,
      "value": "City"
    }
  ],
  "checkout:address-default-address": [
    {
      "type": 0,
      "value": "Make my default address"
    }
  ],
  "checkout:address-email": [
    {
      "type": 0,
      "value": "E-Mail"
    }
  ],
  "checkout:address-firstname": [
    {
      "type": 0,
      "value": "First Name"
    }
  ],
  "checkout:address-form-address-select-new-address": [
    {
      "type": 0,
      "value": "New Address"
    }
  ],
  "checkout:address-form-heading": [
    {
      "type": 0,
      "value": "Shipping Address"
    }
  ],
  "checkout:address-lastname": [
    {
      "type": 0,
      "value": "Last Name"
    }
  ],
  "checkout:address-phone": [
    {
      "type": 0,
      "value": "Phone"
    }
  ],
  "checkout:address-postcode": [
    {
      "type": 0,
      "value": "ZIP"
    }
  ],
  "checkout:address-save-in-address-book": [
    {
      "type": 0,
      "value": "Save in address book"
    }
  ],
  "checkout:address-state": [
    {
      "type": 0,
      "value": "State"
    }
  ],
  "checkout:address-street": [
    {
      "type": 0,
      "value": "Street"
    }
  ],
  "checkout:address-submit": [
    {
      "type": 0,
      "value": "Use Address"
    }
  ],
  "checkout:address-use-saved-address": [
    {
      "type": 0,
      "value": "Use Saved Address"
    }
  ],
  "checkout:back-to-cart": [
    {
      "type": 0,
      "value": "Back to cart"
    }
  ],
  "checkout:checkout": [
    {
      "type": 0,
      "value": "Checkout"
    }
  ],
  "checkout:confirm-order": [
    {
      "type": 0,
      "value": "Confirm Order"
    }
  ],
  "checkout:continue-shopping": [
    {
      "type": 0,
      "value": "Continue Shopping"
    }
  ],
  "checkout:country": [
    {
      "type": 0,
      "value": "Country"
    }
  ],
  "checkout:order-confirmation": [
    {
      "type": 0,
      "value": "The order number is "
    },
    {
      "type": 1,
      "value": "orderId"
    },
    {
      "type": 0,
      "value": ". You will receive an order confirmation email with order status and other details."
    }
  ],
  "checkout:pay-with": [
    {
      "type": 0,
      "value": "Pay With"
    }
  ],
  "checkout:same-as-shipping": [
    {
      "type": 0,
      "value": "Billing address same as shipping address"
    }
  ],
  "checkout:ship-to": [
    {
      "type": 0,
      "value": "Ship To"
    }
  ],
  "checkout:shipping-information": [
    {
      "type": 0,
      "value": "Shipping Information"
    }
  ],
  "checkout:shipping-method": [
    {
      "type": 0,
      "value": "Shipping Method"
    }
  ],
  "checkout:specify-shipping-method": [
    {
      "type": 0,
      "value": "Specify Shipping Method"
    }
  ],
  "checkout:thankyou-for-purchase": [
    {
      "type": 0,
      "value": "Thank you for your purchase!"
    }
  ],
  "checkout:total": [
    {
      "type": 0,
      "value": "TOTAL"
    }
  ],
  "checkout:use": [
    {
      "type": 0,
      "value": "Use"
    }
  ],
  "checkout:use-card": [
    {
      "type": 0,
      "value": "Use Card"
    }
  ],
  "checkout:use-payment-method": [
    {
      "type": 0,
      "value": "Use Payment Method"
    }
  ],
  "checkout:use-shipping-method": [
    {
      "type": 0,
      "value": "Use Method"
    }
  ],
  "common": [
    {
      "type": 0,
      "value": "Edit"
    }
  ],
  "common:back": [
    {
      "type": 0,
      "value": "Back"
    }
  ],
  "common:cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "common:close": [
    {
      "type": 0,
      "value": "Close"
    }
  ],
  "common:loading": [
    {
      "type": 0,
      "value": "Loading"
    }
  ],
  "common:save": [
    {
      "type": 0,
      "value": "Save"
    }
  ],
  "product:add-item": [
    {
      "type": 0,
      "value": "Add to Cart"
    }
  ],
  "product:add-to-wishlist": [
    {
      "type": 0,
      "value": "Add to Wish List"
    }
  ],
  "product:customization-price": [
    {
      "type": 0,
      "value": "Your customization"
    }
  ],
  "product:customize-bundle": [
    {
      "type": 0,
      "value": "Customize"
    }
  ],
  "product:gift-card-amount-selection": [
    {
      "type": 0,
      "value": "Amount"
    }
  ],
  "product:gift-card-choose-amount": [
    {
      "type": 0,
      "value": "Choose an Amount"
    }
  ],
  "product:gift-card-custom-ammount-maximum": [
    {
      "type": 0,
      "value": "Maximum"
    }
  ],
  "product:gift-card-custom-ammount-minimum": [
    {
      "type": 0,
      "value": "Minimum"
    }
  ],
  "product:gift-card-custom-amount": [
    {
      "type": 0,
      "value": "Amount in"
    }
  ],
  "product:gift-card-other-amount": [
    {
      "type": 0,
      "value": "Other Amount"
    }
  ],
  "product:quantity-label": [
    {
      "type": 0,
      "value": "Product quantity"
    }
  ],
  "product:required-fields": [
    {
      "type": 0,
      "value": "Required fields"
    }
  ]
}