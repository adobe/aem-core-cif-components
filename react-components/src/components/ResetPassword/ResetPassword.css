.root {
    padding: 1rem;
}

.fields {
    max-width: 300px;
}

.lead {
    background-color: rgb(var(--venia-grey));
    border-radius: 4px;
    font-size: 0.875rem;
    font-weight: 300;
    line-height: 1.25rem;
    padding: 1rem;
    margin: 1rem 0 1rem 0;
}

.submit {
    padding-top: 1rem;
}

.error {
    color: rgb(var(--venia-error));
    background-color: rgb(var(--venia-grey));
    border-radius: 4px;
    font-size: 0.875rem;
    font-weight: 300;
    line-height: 1.25rem;
    padding: 1rem;
    margin: 1rem 0 1rem 0;
}
