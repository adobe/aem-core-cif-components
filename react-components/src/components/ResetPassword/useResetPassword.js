/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import { useState, useCallback } from 'react';
import { useMutation } from '@apollo/client';

import MUTATION_RESET_PASSWORD from '../../queries/mutation_reset_password.graphql';

const useResetPassword = () => {
    const [status, setStatus] = useState('new');

    const [resetPassword] = useMutation(MUTATION_RESET_PASSWORD);

    const handleFormSubmit = useCallback(
        async ({ email, token, password }) => {
            setStatus('loading');
            try {
                await resetPassword({ variables: { email, resetPasswordToken: token, newPassword: password } });
                setStatus('done');
            } catch (err) {
                setStatus('error');
            }
        },
        [resetPassword]
    );

    return [status, { handleFormSubmit }];
};

export default useResetPassword;
