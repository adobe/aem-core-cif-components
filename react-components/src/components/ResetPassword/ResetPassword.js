/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { Form } from 'informed';
import { useTranslation } from 'react-i18next';

import { useQueryParams } from '../../utils/hooks';
import Field from '../Field';
import TextInput from '../TextInput';
import Button from '../Button';
import combine from '../../utils/combineValidators';
import {
    isRequired,
    validateEmail,
    validatePassword,
    validateConfirmPassword,
    hasLengthAtLeast
} from '../../utils/formValidators';
import useResetPassword from './useResetPassword';
import LoadingIndicator from '../LoadingIndicator';

import classes from './ResetPassword.css';

const ResetPassword = () => {
    const [t] = useTranslation('account', 'common');
    const queryParams = useQueryParams();
    const token = queryParams.get('token');
    const [status, { handleFormSubmit }] = useResetPassword();

    const onSubmit = formValues => {
        const { email, password } = formValues;
        handleFormSubmit({ email, password, token });
    };

    if (!token) {
        return (
            <div className={classes.error}>
                {t('account:reset-password-missing-token', 'Missing or invalid token.')}
            </div>
        );
    }

    if (status === 'loading') {
        return <LoadingIndicator>{t('common:loading', 'Loading')}</LoadingIndicator>;
    }

    if (status === 'error') {
        return <div className={classes.error}>{t('account:reset-password-error', 'Could not reset password.')}</div>;
    }

    if (status === 'done') {
        return (
            <div className={classes.root}>
                <p className={classes.lead}>
                    {t(
                        'account:reset-password-success',
                        'Your password was changed. Please log in with your new password.'
                    )}
                </p>
            </div>
        );
    }

    return (
        <Form className={classes.root} onSubmit={onSubmit}>
            <p className={classes.lead}>
                {t('account:reset-password-lead', 'Please choose a new password to complete the password reset.')}
            </p>
            <div className={classes.fields}>
                <Field label={t('account:email', 'Email address')} required={true}>
                    <TextInput
                        field="email"
                        autoComplete="email"
                        validate={combine([isRequired, validateEmail])}
                        validateOnBlur
                        aria-label="email"
                    />
                </Field>
                <Field label={t('account:new-password', 'New Password')} required={true}>
                    <TextInput
                        field="password"
                        type="password"
                        autoComplete="new-password"
                        validate={combine([isRequired, [hasLengthAtLeast, 8], validatePassword])}
                        validateOnBlur
                        aria-label="password"
                    />
                </Field>
                <Field label={t('account:new-password-confirm', 'Confirm new Password')} required={true}>
                    <TextInput
                        field="confirm"
                        type="password"
                        validate={combine([isRequired, validateConfirmPassword])}
                        validateOnBlur
                        aria-label="confirm"
                    />
                </Field>
            </div>
            <div className={classes.submit}>
                <Button type="submit" priority="high" aria-label="submit">
                    {t('account:reset-password', 'Reset Password')}
                </Button>
            </div>
        </Form>
    );
};

export default ResetPassword;
