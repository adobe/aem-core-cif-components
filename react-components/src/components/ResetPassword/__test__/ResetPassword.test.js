/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MockedProvider } from '@apollo/client/testing';

import MUTATION_RESET_PASSWORD from '../../../queries/mutation_reset_password.graphql';

const mockQueryParams = { get: jest.fn() };
jest.mock('../../../utils/hooks', () => ({
    useQueryParams: () => mockQueryParams
}));

import ResetPassword from '../ResetPassword';
import i18n from '../../../../__mocks__/i18nForTests';

describe('ResetPassword', () => {
    afterEach(() => {
        mockQueryParams.get.mockReset();
    });

    it('renders the form', () => {
        mockQueryParams.get.mockReturnValue('mytoken');
        const { asFragment } = render(
            <I18nextProvider i18n={i18n}>
                <MockedProvider>
                    <ResetPassword />
                </MockedProvider>
            </I18nextProvider>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders an error message for a missing token', () => {
        mockQueryParams.get.mockReturnValue(null);
        const { asFragment } = render(
            <I18nextProvider i18n={i18n}>
                <MockedProvider>
                    <ResetPassword />
                </MockedProvider>
            </I18nextProvider>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders an error message for a failed password reset', async () => {
        mockQueryParams.get.mockReturnValue('mytoken');
        const mocks = [
            {
                request: {
                    query: MUTATION_RESET_PASSWORD,
                    variables: {
                        email: 'chuck@example.com',
                        resetPasswordToken: 'mytoken',
                        newPassword: 'NewPassword123'
                    }
                },
                result: {
                    data: { resetPassword: false },
                    errors: [{ message: 'Cannot set the customers password' }]
                }
            }
        ];

        const { asFragment, getByLabelText, getByText } = render(
            <I18nextProvider i18n={i18n}>
                <MockedProvider mocks={mocks} addTypename={false}>
                    <ResetPassword />
                </MockedProvider>
            </I18nextProvider>
        );

        fireEvent.change(getByLabelText('email'), { target: { value: 'chuck@example.com' } });
        fireEvent.change(getByLabelText('password'), { target: { value: 'NewPassword123' } });
        fireEvent.change(getByLabelText('confirm'), { target: { value: 'NewPassword123' } });
        fireEvent.click(getByLabelText('submit'));

        await waitForElement(() => getByText('Could not reset password.'));
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders a success message', async () => {
        mockQueryParams.get.mockReturnValue('mytoken');
        const mocks = [
            {
                request: {
                    query: MUTATION_RESET_PASSWORD,
                    variables: {
                        email: 'chuck@example.com',
                        resetPasswordToken: 'mytoken',
                        newPassword: 'NewPassword123'
                    }
                },
                result: {
                    data: { resetPassword: true }
                }
            }
        ];

        const { asFragment, getByLabelText, getByText } = render(
            <I18nextProvider i18n={i18n}>
                <MockedProvider mocks={mocks} addTypename={false}>
                    <ResetPassword />
                </MockedProvider>
            </I18nextProvider>
        );

        fireEvent.change(getByLabelText('email'), { target: { value: 'chuck@example.com' } });
        fireEvent.change(getByLabelText('password'), { target: { value: 'NewPassword123' } });
        fireEvent.change(getByLabelText('confirm'), { target: { value: 'NewPassword123' } });
        fireEvent.click(getByLabelText('submit'));

        await waitForElement(() => getByText('Your password was changed. Please log in with your new password.'));
        expect(asFragment()).toMatchSnapshot();
    });
});
