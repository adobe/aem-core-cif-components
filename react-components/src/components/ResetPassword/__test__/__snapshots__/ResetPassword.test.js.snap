// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword renders a success message 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <p
      class="lead"
    >
      Your password was changed. Please log in with your new password.
    </p>
  </div>
</DocumentFragment>
`;

exports[`ResetPassword renders an error message for a failed password reset 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    Could not reset password.
  </div>
</DocumentFragment>
`;

exports[`ResetPassword renders an error message for a missing token 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    Missing or invalid token.
  </div>
</DocumentFragment>
`;

exports[`ResetPassword renders the form 1`] = `
<DocumentFragment>
  <form
    class="root"
  >
    <p
      class="lead"
    >
      Please choose a new password to complete the password reset.
    </p>
    <div
      class="fields"
    >
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Email address
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="email"
              autocomplete="email"
              class="input"
              name="email"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          New Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="password"
              autocomplete="new-password"
              class="input"
              name="password"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Confirm new Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="confirm"
              class="input"
              name="confirm"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
    </div>
    <div
      class="submit"
    >
      <button
        aria-label="submit"
        class="root_highPriority"
        type="submit"
      >
        <span
          class="content"
        >
          Reset Password
        </span>
      </button>
    </div>
  </form>
</DocumentFragment>
`;
